pkg/kubelet/cm/internal_container_lifecycle.go

```diff
@@ -37,13 +38,23 @@ type internalContainerLifecycleImpl struct {
 	topologyManager topologymanager.Manager
 }
 
+func (i *internalContainerLifecycleImpl) PreCreateContainer(pod *v1.Pod, container *v1.Container, containerConfig *runtimeapi.ContainerConfig) error {
+	if containerConfig.Linux != nil && containerConfig.Linux.Resources != nil {
+		if i.cpuManager != nil {
+			allocatedCPUs := i.cpuManager.GetAllocatedResources(pod, container)
+			if allocatedCPUs != "" {
+				containerConfig.Linux.Resources.CpusetMems = allocatedCPUs
+			}
+		}
+	}
```

Comment:
I don't think we simply want to check for `containerConfig.Linux != nil && containerConfig.Linux.Resources != nil` and then set CPUs in them if they exist. I think we want to explicitly create those structures if they do not exist and then add CPUs to them. Moreover, this should explicitly only be done on linux, so something like:
```
if i.cpuManager == nil {
    return nil
}

if runtime.GOOS != "linux" {
    return nil
}

allocatedCPUs := i.cpuManager.GetAllocatedResources(pod, container)
if allocatedCPUs == "" {
    return nil
}

if containerConfig.Linux == nil {
    containerConfig.Linux = ...
}

if containerConfig.Linux.Resources == nil {
    containerConfig.Linux.Resources = ...
}

containerConfig.Linux.Resources.CpusetMems = allocatedCPUs

return nil
```

Reply:
I checked again the code and I can see that all needed structure initialized under https://github.com/kubernetes/kubernetes/blob/3b38f6a08e255b3eb2e34ffb3c3cecacc081f334/pkg/kubelet/kuberuntime/kuberuntime_container.go#L160 so I will just remove all redundant checks.